src/microvm/src/memory_thread.rs

```diff
@@ -79,6 +93,32 @@ where
                     // No message available.
                 },
             }
+            match control_rx.try_recv() {
+                Ok(MemoryControlCommand::Pause) => {
+                    trace!("pause()");
+                    crate::timer!("vm_pause");
+                    if pause_microvm().is_err() {
+                        control_tx.send(MemoryControlResponse::PauseError)?;
+                    }
+                },
+                Ok(MemoryControlCommand::Resume) => {
+                    trace!("resume()");
+                    crate::timer!("vm_resume");
+                    if resume_microvm().is_err() {
+                        control_tx.send(MemoryControlResponse::ResumeError)?;
+                    }
+                    control_tx.send(MemoryControlResponse::ResumeWritten)?;
+                },
+                Err(TryRecvError::Disconnected) => {
+                    // When the guest finishes , the vCPU thread will disconnect from this
+                    // thread. This situation is normal and should not create an error log.
+                    debug!("spawn(): channel has been disconnected");
+                    break Ok(());
+                },
+                Err(TryRecvError::Empty) => {
+                    // No message available.
+                },
+            }
```

Comment:
Trace tags should contain the name of the function. For this case, all trace statements should use "spawn()". Rationale: this is used to pin-point the module and function where the log comes from. Since in the same module function names are unique, there is no way of having any ambiguity.